src/components/resource/components/edit/coverage-settings/coverage-settings.test.js

```diff
@@ -0,0 +1,153 @@
+          customCoverages: (props.model || defaultModel).customCoverages,
+        }}
+        render={({ pristine, form: { reset } }) => (
+          <form onSubmit={onSubmitMock}>
+            <EditCoverageSettings
+              isOpen
+              model={defaultModel}
+              resourceSelected
+              getSectionHeader={() => 'Section label'}
+              handleSectionToggle={handleSectionToggleMock}
+              {...props}
+            />
+            <button type="button" disabled={pristine} onClick={reset}>Cancel</button>
+            <button type="submit" disabled={pristine}>Submit</button>
+          </form>
+        )}
+      />
+    </Harness>
+  );
+
+  afterEach(() => {
+    cleanup();
+    handleSectionToggleMock.mockClear();
+  });
+
+  it('should render an accordion', () => {
+    const { getByText } = renderCoverageSettings();
+    expect(getByText('Section label')).toBeDefined();
+  });
+
+  it('should render CoverageDateList', () => {
+    const { getAllByText } = renderCoverageSettings();
+    expect(getAllByText('Coverage date list')).toBeDefined();
+  });
+
+  describe('when resource is not selected', () => {
+    it('should render message', () => {
+      const { getByText } = renderCoverageSettings({
+        resourceSelected: false,
+      });
+
+      expect(getByText('ui-eholdings.resource.coverage.notSelected')).toBeDefined();
+    });
+  });
+
+  describe('when there are no coverage dates', () => {
+    it('should not render coverage date list', () => {
+      const { queryByText } = renderCoverageSettings({
+        model: {
+          ...defaultModel,
+          customCoverages: [],
+          managedCoverages: [],
+        },
+      });
+
+      expect(queryByText('Coverage date list')).toBeNull();
+    });
+  });
+
+  // describe('when editing and saving coverage dates', () => {
+  //   it('should call onSubmit', () => {
+  //     const {
+  //       getByLabelText,
+  //       getByText,
+  //     } = renderCoverageSettings();
+
+  //     const startDateInput = getByLabelText('ui-eholdings.date.startDate');
+  //     const endDateInput = getByLabelText('ui-eholdings.date.endDate');
+
+  //     fireEvent.change(startDateInput, { target: { value: '01/01/2021' } });
+  //     fireEvent.change(endDateInput, { target: { value: '01/21/2021' } });
+  //     fireEvent.click(getByText('Submit'));
+  //     expect(onSubmitMock).toBeCalled();
+  //   });
+  // });
+
+  // describe('when setting incorrect coverage dates', () => {
+  //   it('should show error message', () => {
+  //     const {
+  //       getByLabelText,
+  //       getByText,
+  //     } = renderCoverageSettings();
+
+  //     const startDateInput = getByLabelText('ui-eholdings.date.startDate');
+  //     const endDateInput = getByLabelText('ui-eholdings.date.endDate');
+
+  //     fireEvent.change(startDateInput, { target: { value: '01/21/2021' } });
+  //     fireEvent.change(endDateInput, { target: { value: '01/01/2021' } });
```

Comment:
Probably you need use `blur` for the test to pass, try
```
fireEvent.blur(endDateInput);
```

Reply:
That worked, thanks!